test/std/algorithms/alg.sorting/alg.binary.search/lower.bound/partitioned_g.pass.cpp

```diff
@@ -0,0 +1,187 @@
+    }
+    // check if there is change after executing kernel function
+    check &= check_data(tmp, seq, N);
+    if (!check)
+        return false;
+    return ret;
+}
+
+struct Y
+{
+    double val;
+
+    // Not irreflexive, so not a strict weak order.
+    bool
+    operator<(const Y& y) const
+    {
+        return val < int(y.val);
+    }
+    bool
+    operator!=(const Y& y) const
+    {
+        return val != (y.val);
+    }
+};
+
+bool
+kernel_test2(sycl::queue& deviceQueue)
+{
+    Y seq[] = {-0.1, 1.2, 5.0, 5.2, 5.1, 5.9, 5.5, 6.0};
+    auto tmp = seq;
+    const int N = sizeof(seq) / sizeof(seq[0]);
+    bool ret = false;
+    bool check = false;
+    sycl::range<1> item1{1};
+    sycl::range<1> itemN{8};
+    {
+        sycl::buffer<bool, 1> buffer1(&ret, item1);
+        sycl::buffer<bool, 1> buffer2(&check, item1);
+        sycl::buffer<Y, 1> buffer3(seq, itemN);
+        deviceQueue.submit([&](sycl::handler& cgh) {
+            auto ret_access = buffer1.get_access<sycl_write>(cgh);
+            auto check_access = buffer2.get_access<sycl_write>(cgh);
+            auto access = buffer3.get_access<sycl_write>(cgh);
+            cgh.single_task<class KernelTest2>([=]() {
+                Y arr[] = {-0.1, 1.2, 5.0, 5.2, 5.1, 5.9, 5.5, 6.0};
+                // check if there is change after data transfer
+                check_access[0] = check_data(&access[0], arr, N);
+                if (check_access[0])
+                {
+                    test_container<Y, forward_iterator_wrapper> c(&access[0], &access[0] + N);
+                    auto part1 = std::lower_bound(c.begin(), c.end(), Y{5.5});
+                    ret_access[0] = (part1 != c.end());
+                    ret_access[0] &= (part1->val == 5.0);
+                    auto part2 = std::lower_bound(c.begin(), c.end(), Y{5.5}, std::less<Y>{});
+                    ret_access[0] &= (part2 != c.end());
+                    ret_access[0] &= (part2->val == 5.0);
+
+                    auto part3 = std::lower_bound(c.begin(), c.end(), Y{1.0});
+                    ret_access[0] &= (part3 != c.end());
+                    ret_access[0] &= (part3->val == 1.2);
+                    auto part4 = std::lower_bound(c.begin(), c.end(), Y{1.0}, std::less<Y>{});
+                    ret_access[0] &= (part4 != c.end());
+                    ret_access[0] &= (part4->val == 1.2);
+                }
+            });
+        }).wait();
+    }
+    // check if there is change after executing kernel function
+    check &= check_data(tmp, seq, N);
+    if (!check)
+        return false;
+    return ret;
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    sycl::queue deviceQueue = TestUtils::get_test_queue();
+    auto ret = kernel_test1(deviceQueue);
+    if (TestUtils::has_type_support<double>(deviceQueue.get_device()))
+    {
+        ret &= kernel_test2(deviceQueue);
+    }
+    EXPECT_TRUE(ret, "Wrong result of lower_bound");
```

Comment:
What's a difference form others "lower_bound" tests?
I think a diff should be mentioned in this error message as well (to quick get info and find place).

Reply:
Fixed